src/libos/src/config.rs

```diff
@@ -446,4 +493,131 @@ struct InputConfigMountOptions {
     pub layers: Option<Vec<InputConfigMount>>,
     #[serde(default)]
     pub temporary: bool,
+    #[serde(default)]
+    pub index: u32,
+    #[serde(default)]
+    pub autokey_policy: Option<u32>,
+}
+
+#[derive(Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+struct InputConfigApp {
+    #[serde(default)]
+    pub stage: String,
+    #[serde(default)]
+    pub entry_points: Vec<String>,
+    #[serde(default)]
+    pub mount: Vec<InputConfigMount>,
+    #[serde(default)]
+    pub encrypted: bool,
+}
+
+#[repr(C)]
+#[derive(Debug, Copy, Clone)]
+#[allow(non_camel_case_types)]
+pub struct user_rootfs_config {
+    upper_layer_path: *const i8,
+    lower_layer_path: *const i8,
+    entry_point: *const i8,
+    hostfs_source: *const i8,
+    hostfs_target: *const i8,
+}
+
+impl user_rootfs_config {
+    pub fn from_raw_ptr(ptr: *const user_rootfs_config) -> Result<user_rootfs_config> {
```

Comment:
this `from_raw_ptr ` method should be unsafe, and it should not return `Result<T>`, because this method cannot fail.